### PR TITLE
(WIP) x86-64 JIT

### PR DESCRIPTION
**Pull request by Alegend45**
Please do not merge this yet. There's so much more work to do on this PR it's not even funny. Feel free to review what I've got so far though. I fear licensing comments may be an issue.

### CODE REVIEW

**Comment by archshift** [comment]
Also note that endianness conversions are not an issue here, whereas they are with PPC.

**Comment by Alegend45** [comment]
Ah, okay. I know ARM's bi-endian, so it's good to know that in the case of the 3DS I only have to support little-endian stuff. :P

**Comment by neobrain** [comment]
Hi,

Just a quick note - we usually don't use WIP PRs unless they are close to being finalized (in which case they should be labelled RFC, i.e. Request For Comments). If everyone were to submit whatever branch they are currently working on as a PR, the PR list would be pointlessly cluttered and review would become too confusing.

**Comment by lioncash** [comment]
This should also ideally be done after the interpreter is done being implemented, since you have a reference to compare against in terms of correctness.

**Comment by Alegend45** [comment]
Oh, okay, sorry @neobrain I was going off of Dolphin conventions.

@lioncash True, but I'm at least going to get the basics done and have everything else fall back to interpreter.

**Comment by lioncash** [comment]
If you plan to implement instructions in the JIT, I'd like if you provided emitter tests for each instruction as well in order to verify that the emission is indeed correct.

**Comment by Alegend45** [comment]
Alright, will do.

**Comment by Alegend45** [comment]
Alright, I've got a JIT up and running that always falls back to dyncom, lemme just test it with blargSnes real quick just to make sure I got it right.

File: src/core/arm/jit_common/jit_base.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
+// Copyright 2014 Citra Emulator Project
```
Comment (purpasmart96) [inline src/core/arm/jit_common/jit_base.cpp]: Update the licenses to this along with the other files you added which are licensed under Citra
// Copyright 2015 Citra Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.

**Comment by Alegend45** [comment]
I can't freaking figure out these build problems. I don't do a lot of inheritance stuff.

**Comment by yuriks** [comment]
I commented about this in IRC, but maybe we should look into basing our JIT on a light-weight SSA representation. This should make it more straight-forward to implement better optimizations later, as well as being (conceptually) simpler to understand code.

**Comment by Alegend45** [comment]
Yeah, I guess this should wait, given the fact that software rendering is the main bottleneck according to you guys.

**Pull request closed without merging**